ambry-frontend/src/main/java/com.github.ambry.frontend/PostBlobHandler.java

```diff
@@ -34,30 +41,43 @@
  * Handler for post blob requests.
  */
 class PostBlobHandler {
-
   private static final Logger LOGGER = LoggerFactory.getLogger(PostBlobHandler.class);
+  /**
+   * Key to represent the time at which a blob will expire in ms. Used within the metadata map in signed IDs.
+   */
+  static final String EXPIRATION_TIME_MS_KEY = "et";
 
   private final SecurityService securityService;
   private final IdConverter idConverter;
+  private final Router router;
   private final AccountAndContainerInjector accountAndContainerInjector;
+  private final Time time;
+  private final FrontendConfig frontendConfig;
+  private final int chunkUploadMaxChunkSize;
   private final FrontendMetrics frontendMetrics;
-  private final Router router;
 
   /**
    * Constructs a handler for handling requests for signed URLs.
    * @param securityService the {@link SecurityService} to use.
    * @param idConverter the {@link IdConverter} to use.
-   * @param accountAndContainerInjector helper to resolve account and container for a given request.
-   * @param frontendMetrics {@link FrontendMetrics} instance where frontendMetrics should be recorded.
    * @param router the {@link Router} to use.
+   * @param accountAndContainerInjector helper to resolve account and container for a given request.
+   * @param time the {@link Time} instance to use.
+   * @param frontendConfig the {@link FrontendConfig} to use.
+   * @param chunkUploadMaxChunkSize the preconfigured max size for chunks of a stitched upload.
+   * @param frontendMetrics {@link FrontendMetrics} instance where metrics should be recorded.
    */
-  PostBlobHandler(SecurityService securityService, IdConverter idConverter,
-      AccountAndContainerInjector accountAndContainerInjector, FrontendMetrics frontendMetrics, Router router) {
+  PostBlobHandler(SecurityService securityService, IdConverter idConverter, Router router,
+      AccountAndContainerInjector accountAndContainerInjector, Time time, FrontendConfig frontendConfig,
+      int chunkUploadMaxChunkSize, FrontendMetrics frontendMetrics) {
     this.securityService = securityService;
     this.idConverter = idConverter;
+    this.router = router;
     this.accountAndContainerInjector = accountAndContainerInjector;
+    this.time = time;
+    this.frontendConfig = frontendConfig;
+    this.chunkUploadMaxChunkSize = chunkUploadMaxChunkSize;
```

Comment:
i think we should remove this from here - creates too tight a coupling b/w this and the router impl